### ISSUE TITLE
IntuneMAMComplianceDelegate with AutoEnrollOnLaunch

### ISSUE THREAD

**Issue by dmholmes** [microsoftconnect/ms-intune-app-sdk-ios#483]
Is there a way to get notified when authentication is complete when using AutoEnrollOnLaunch set to true for app protection?

**Comment by andrejacobs-unily** [comment]
Yes also please let us know how we can get notified when the Intune MAM SDK has finished enrollment and is happy and when the MAM SDKs view are being dismissed etc. #487

@dmholmes Not sure if you are looking for the same thing as me. You can get a bit more information about the enrollment and policy status if you hookup a IntuneMAMEnrollmentDelegate to `IntuneMAMEnrollmentManager.instance().delegate`. 
I have managed to run this in our application didFinishLaunchingWithOptions and the delegate methods do get called. More info here: https://learn.microsoft.com/en-us/mem/intune/developer/app-sdk-ios-phase3#status-result-and-debug-notifications

**Comment by dmholmes** [comment]
@andrejacobs-unily the delegate methods (only looking for hasComplianceStatus at the moment) are not being called. I suspect it's because we are letting Intune handle auth at launch (https://learn.microsoft.com/en-us/mem/intune/developer/app-sdk-ios-phase3#let-intune-handle-authentication-and-enrollment-at-launch).

**Comment by kanishkaBagga** [comment]
@andrejacobs-unily - we are investigating and working on adding notifications that inform applications of when the Intune SDK has taken or relinquished UI control. This should be available soon once completed with the work
